### DISCUSSION TITLE
Rdf output support

### DISCUSSION THREAD

**Discussion by DosWorld** [samiam95124/Pascal-P6#175] [General]
Hello!

I want purpose idea - support output into rdf object file format.
it will allow produce many type of executable files for ms-dos.
https://github.com/DosWorld/rtools

**Comment by samiam95124** [comment]
Sounds like a great project, I invite you to do it.

First, I doubt it will happen, and I''l explain why momentarily. That is why I moved this to discussions. It can always be moved back, by the way.

Ok, lets go through that.

First, Pascal-P6 is entirely output format agnostic. Meaning it does not control what output format gets generated in the end. It generates assembly code for its modules, and uses C for support files. This means if the compiler/assember you use as a back end for it generates xyz format, then Bob is your uncle, you are there.

Now we get to the hard problem. Its been proposed before to do something like a FreeDOS backend. I don't have an issue with that, I think it would be great. But DOS is 16 bit, with 32 bit extensions via DOS extenders. The backend for Pascal-P6 with respect for x86 processors is 64 bit. Nobody I know has a 64 bit DOS extender, although one is certainly possible. The other way, of course, is to make a 32 bit x86 pgen for Pascal-P6.

Who would do that? pgen is really easy to write backends for.  However, my realistic priorities for next CPUs/models to support are:

ARM 64 bit (Apple Macs)
ARM 32 bit (embedded)
RISC-V 32 bit
RISC-V 64 bit

In that order. Why? Those are the most common CPU dependencies going forward. 

Scott
